backend/src/main/java/me/performancereservation/domain/auth/controller/AuthController.java

```diff
@@ -0,0 +1,41 @@
+package me.performancereservation.domain.auth.controller;
+
+import lombok.RequiredArgsConstructor;
+import me.performancereservation.domain.user.entitiy.User;
+import me.performancereservation.domain.user.enums.Role;
+import me.performancereservation.domain.user.service.UserService;
+import me.performancereservation.global.exception.AppException;
+import me.performancereservation.global.security.jwt.JwtTokenProvider;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("api/v1/auth")
+@RequiredArgsConstructor
+public class AuthController {
+
+    private final UserService userService;
+    private final JwtTokenProvider jwtTokenProvider;
+
+    //이메일로 회원가입 + 토큰 발급을 테스트
+    @PostMapping("/signup-test")
+    public ResponseEntity<String> testSignUp(
+            @RequestParam String email, @RequestParam String name) {
+        User user;
+        try {
+            user = userService.getUserByEmail(email);
+        } catch (AppException e) {
+            user = userService.registerUser(email,name, null, Role.USER);
+        }
```

Comment:
컨트롤러 계층에서 예외를 발생 시키는 것 보단 서비스 계층에서 처리하는 방식이 좋지 않을까요? 컨트롤러는 Http에 대한 처리만 수행하고 서비스 계층에서 이렇게 처리해 보는 건 어떨지 제안 드려봅니다!
```suggestion
        User user = userService.getOrCreateUser(email, name);
        
        // UserService
        public User getOrCreateUser(String email, String name) {
               return userRepository.findByEmail(email)
                           .orElseGet(() -> registerUser(email, name, null, Role.USER));
        }
        
```

Reply:
확실히 예외는 서비스 계층에서 처리하는 것이 좋다고 생각합니다!!
삭제할 내용이라 신경쓰지 못했던 것 같습니다. 수정하고 오겠습니다 :)

Reply:
아 테스트 용으로만 쓰면 상관 없을 것 같습니다. 코드에만 꽂혀서 잘 확인을 못했네요!

Reply:
대충 보고 넘어갈 수 있었던 부분이였는데 찾아주셔서 감사합니다 ㅎㅎ
그래도 기본은 지키는게 좋겠다는 생각이 들었습니다 후딱 고쳐놓았습니다 😎